### ISSUE TITLE
Private frameworks are built when doing `carthage build --no-skip-current`

### ISSUE THREAD

**Issue by eimantas** [Carthage/Carthage#2612]
* carthage install method: [x] .pkg, [ ] homebrew, [ ] source
* `which carthage`: /usr/local/bin/carthage
* `carthage version`: 0.31.0
* `xcodebuild -version`: Xcode 10.0; Build version 10A255
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No
* Are you using `--new-resolver`? No

**Cartfile**

```
```

**Cartfile.private**
```
github "quick/quick" ~> 1.3
github "quick/nimble" ~> 7.3
```

**Carthage Output**
```
% carthage build --no-skip-current --platform iOS
*** xcodebuild output can be found in /var/folders/c8/sg3q64zj1tqg5bcqrqzzp96m0000gn/T/carthage-xcodebuild.BUY74z.log
*** Building scheme "Nimble-iOS" in Nimble.xcodeproj
*** Building scheme "Quick-iOS" in Quick.xcworkspace
*** Building scheme "MyFramework" in MyFramework.xcodeproj
```

**Actual outcome**
Carthage built private (development) frameworks for all platforms when testing if framework will build for client.

**Expected outcome**
Carthage should not have built those as they are needed only when running specs.

**Comment by blender** [comment]
Private just means that they won’t be added as transitive dependencies. Carthage has no concept of testing or relation ship between dependencies and targets. This is intended. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Dopebiz6Dime** [comment]
Sorry phone I had that was working with it was android take over ... lost everything 

Daniel L Wotapka

On Nov 12, 2018, at 6:53 PM, Matt Diephouse <notifications@github.com> wrote:

Closed #2612.
